src/components/interactive-builder/add-question.modal.tsx

```diff
@@ -139,9 +131,33 @@ const AddQuestionModal: React.FC<AddQuestionModalProps> = ({
     programState: ['select'],
   };
 
+  const datePickerTypeOptions: DatePickerTypeOptions = {
+    datetime: [{ value: 'both', label: t('calendarAndTimer', 'Calendar and timer'), defaultChecked: true }],
+    date: [{ value: 'calendar', label: t('calendarOnly', 'Calendar only'), defaultChecked: false }],
+    time: [{ value: 'timer', label: t('timerOnly', 'Timer only'), defaultChecked: false }],
+  };
+
   const handleConceptChange = (event: React.ChangeEvent<HTMLInputElement>) => setConceptToLookup(event.target.value);
 
+  const updateDatePickerType = (concept: Concept) => {
+    const conceptDataType = concept.datatype.name;
+    switch (conceptDataType) {
+      case 'Datetime':
+        setDatePickerType('both');
+        break;
+      case 'Date':
+        setDatePickerType('calendar');
+        break;
+      case 'Time':
+        setDatePickerType('timer');
+        break;
+      default:
+        break;
```

Comment:
Can we handle the default case? Maybe log some kind of error or warning message:

```ts
default:
      console.warn(`Unsupported datatype for date fields: ${conceptDataType}`);
      break;
```